pkg/plugin/plugin.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/cs3org/reva/pkg/pluginregistry"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+)
+
+var handshake = plugin.HandshakeConfig{
+	ProtocolVersion:  1,
+	MagicCookieKey:   "BASIC_PLUGIN",
+	MagicCookieValue: "hello",
+}
+
+func compile(path string, pluginType string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("could not find current directory: %v", err)
+	}
+	pluginsDir := filepath.Join(wd, "bin", pluginType, filepath.Base(path))
+	command := fmt.Sprintf("GO111MODULE=off go build -o %s %s", pluginsDir, path)
```

Comment:
Here, the `path` should be *relative*, so when we are configuring the driver, the path should be relative to the current working directory because `go build` does not work when we provide absolute path. So config should be like:
```
[grpc.services.userprovider]
driver = "../json"
```

Reply:
`go build -o json /home/path/to/source` is not supported